### ISSUE TITLE
Does `startWith` work?

### ISSUE THREAD

**Issue by fuyu** [ReactiveKit/Bond#242]
I use a simple code to test:

```
let test = Observable(5)
test.startWith(6).observe { print("\($0)") }
```

I see 5 in logs, but never 6. Can you advice what is wrong with it?

**Comment by goodwithgit** [comment]
I was experiencing a similar issue ... the startWith() event was not seen.

I tried out @yanamura3's fix (in commit 03693b6) for this and noticed that the startWith() event is propagates, but it doesn't do so until a subsequent event is posted.   In my case, I was expecting it to propagate immediately.    Am I wrong to expect that?

I do get the behaviour I want by altering @yanamura3's startWith() slightly as follows:
```
	public func startWith(event: EventType) -> EventProducer<EventType> {
		var hasDispatch = false
		return EventProducer(replayLength: replayLength + 1, lifecycle: EventProducerLifecycle.Managed) { sink in
			if !hasDispatch {
				hasDispatch = true
				sink(event)
			}
			return observe { subsequentEvent in
				sink(subsequentEvent)
			}
		}
	}
```

Is this good or bad?   I'm quite new to Swift, binding, reactive-ness, MVVM, etc. etc., so I'm at a bit of loss to evaluate.

**Comment by srdanrasic** [comment]
This was fixed at one point.
